pkg/kubelet/dockershim/libdocker/fake_client.go

```diff
@@ -145,6 +149,24 @@ func (f *FakeDockerClient) appendContainerTrace(traceCategory string, containerN
 	}
 }
 
+// WriteDockerConfigFile creates a config file that NewDockerKeyring will pick up. It returns
+// a function that will delete the file, so the caller can `defer` it.
+func (f *FakeDockerClient) WriteDockerConfigJSONFile(cfg string) (error, func()) {
```

Comment:
I have some reservations about storing the test config.json file in the current working directory. I think it'd be much better to ensure correctness and proper cleanup by creating a temporary file. To do this, I think we can factor out the code to be a new function that accepts a list of paths to search for docker configs. The test can then create a file in a temp dir and pass it to the function to be tested. 

You won't need this WriteDockerConfigJSONFile function anymore.